version2/aks-cluster.tf

```diff
@@ -0,0 +1,64 @@
+resource "random_pet" "prefix" {}
+
+provider "azurerm" {
+  subscription_id = var.subscription_id
+  client_id       = var.client_id
+  client_secret   = var.client_secret
+  tenant_id       = var.tenant_id
+  version = "~> 2.0"
+  features {}
+}
+
+resource "azurerm_resource_group" "default" {
+  name     = "${random_pet.prefix.id}-rg"
+  location = "West US 2"
+
+  tags = {
+    environment = "Demo"
+  }
+}
+
+resource "azurerm_kubernetes_cluster" "default" {
+  name                = "${random_pet.prefix.id}-aks"
+  location            = azurerm_resource_group.default.location
+  resource_group_name = azurerm_resource_group.default.name
+  dns_prefix          = "${random_pet.prefix.id}-k8s"
+
+  default_node_pool {
+    name            = "default"
+    node_count      = 1
+    vm_size         = "Standard_D2_v2"
+    os_disk_size_gb = 30
+  }
+
+  linux_profile {
+      admin_username = "k8s"
+
+      ssh_key {
+          key_data = file(var.ssh_public_key)
+      }
+  }
+  service_principal {
+    client_id     = var.appId
+    client_secret = var.password
+  }
+
+  role_based_access_control {
+    enabled = true
+  }
+
+  addon_profile {
+      oms_agent {
+      enabled     = false
+      }
+  }
+
+  network_profile {
+    load_balancer_sku = "Standard"
+    network_plugin = "kubenet"
+  }
```

Comment:
I believe these (`addon_profile` and `network_profile`) are the default for the AKS cluster, let's omit them?

Reply:
Changed Thanks